images/cilium/Dockerfile

```diff
@@ -3,10 +3,11 @@
 
 ARG CILIUM_BUILDER_IMAGE=quay.io/cilium/cilium-builder:57042c9f19ec861e3066ecccd3aecae2591791ab@sha256:d6b91c648d572c9503b3f1327614c8c2cfb040167bea71b4ab937cf9676fe3df
 ARG CILIUM_RUNTIME_IMAGE=quay.io/cilium/cilium-runtime:abf6badae18e61009c5d8ae69f3b87866259a470@sha256:d06d8a35e05314bfbf31349ffcca5751dc9678ab463ed1a4376732a1b16bd77e
+ARG CILIUM_ENVOY_IMAGE=quay.io/cilium/cilium-envoy:v1.27.2-f19708f3d0188fe39b7e024b4525b75a9eeee61f@sha256:80de27c1d16ab92923cc0cd1fff90f2e7047a9abf3906fda712268d9cbc5b950
```

Comment:
One of the reasons that it's not passed as argument is to make sure that the helm chart is in sync with envoy deamonset image. There is also a Makefile target (e.g. make -C image check-envoy-image) to validate the same.

Just curious on your use case for this PR, also do you need to install Cilium with one extra flag (e.g. disable-envoy-version-check) ?

Reply:
There are envoy update scripts that deal with the `FROM` line that need to be adjusted for the `ARG`, I recall Makefiles may do that too.

Reply:
Thanks for pointing this out, I hadn't realised.

So:
* Updated `{update,check}-cilium-envoy-image.sh` to patch the Dockerfile ARG entry instead of the FROM.
* Updated the link-time flags to pull the Envoy image ref from the ARG entry instead of the FROM.

Hope I haven't missed anything.

Reply:
Yes, I believe the above will be enough to make CI happy.

Reply:
> Just curious if you have other use case for custom builds, or just for mirrored image and air-gap environment, 

We do custom builds because we typically carry a few patches on top of upstream: mostly these are bug fixes backported from more recent versions or `main`, and occasionnally experiments/dev that hasn't made it upstream yet.

> also do you need to install Cilium with one extra flag (e.g. disable-envoy-version-check) ?

From my understanding that check (`checkEnvoyVersion()`) only verifies the SHA which is unchanged so that option is not needed.